extension-style/src/main/java/com/mapbox/maps/extension/style/sources/Source.kt

```diff
@@ -129,6 +133,17 @@ abstract class Source(
    */
   companion object {
     private const val TAG = "Mbgl-Source"
+
+    private val workerThread by lazy {
+      HandlerThread("STYLE_WORKER").apply {
```

Comment:
Once this thread is started, it will remain running until the whole app process is killed. Just something to consider.

Reply:
+1 on @LukasPaczos remark and additionally want to note that for each source that has lazy initialized it we will have a dedicated workerthread for it. I would prefer using a single - reusable that is managed as part of the MapView lifecycle.

Reply:
I like an idea to introduce a worker thread abstraction backing up given `MapView`. It may be started automatically in `onStart` or lazily and get killed in `onDestroy`. In future we could introduce thread pool implementing given abstraction.

Reply:
I was thinking about that as well, but tieing the lifecycle of that thread to a `MapView` might be harsh.

The `Source` object itself is in no way tied to the `MapView` lifecycle. It could be created before `MapView` is inflated and then added to it. If the thread is tied to the `MapView` for practical reasons, then the `Source` impl has to have some kind of a queue (coalesced) waiting for the thread to get started.

Also a note on the thread pool - it would have to be well synchronized within a source instance, otherwise, the updates might be competing and overwriting each other.